electric_emission_cost/costs.py

```diff
@@ -568,7 +621,8 @@ def calculate_energy_cost(
     if model is None:
         n_steps = consumption_data.shape[0]
     else:  # Pyomo does not support shape attribute
-        n_steps = len(consumption_data.extract_values())
```

Comment:
Did `extract_values()` throw an error or do you just think it is unnecessary?

Reply:
I believe it did.